prime-router/src/main/kotlin/history/azure/ReportFileFunction.kt

```diff
@@ -190,6 +204,75 @@ abstract class ReportFileFunction(
         }
     }
 
+    /**
+     * Function to return metadata of a single report from the CDC Intermediary.
+     * The [id] is a valid report UUID. This function is for the Intermediary only, please don't update
+     * without contacting that engineering team
+     */
+    fun retrieveMetadata(
+        request: HttpRequestMessage<String?>,
+        id: String,
+        context: ExecutionContext,
+        engine: HttpClientEngine?,
+    ): HttpResponseMessage {
+        val authResult = this.authSingleBlocks(request, id)
+
+        if (authResult != null) {
+            return authResult
+        }
+
+        var response: HttpResponse?
+        val receiver = workflowEngine.settings.findReceiver(this.intermediaryReceiverName)
+        val client = if (engine == null) HttpClient() else HttpClient(engine)
+        val restTransportInfo = receiver?.transport as RESTTransportType
+        val (credential, jksCredential) = RESTTransport().getCredential(restTransportInfo, receiver)
+        val logger: Logger = context.logger
+        var authPair: Pair<Map<String, String>?, String?> =
+            Pair(null, null)
+
+        var responseBody = ""
+
+        runBlocking {
+            launch {
+                authPair = RESTTransport().getOAuthToken(
+                    restTransportInfo,
+                    id,
+                    jksCredential,
+                    credential,
+                    logger
+                )
+            }
+        }
+
+        val lookupId = this.getLookupId(id)
+        if (lookupId.isEmpty()) {
+            return HttpUtilities.notFoundResponse(request, "lookup Id not found")
+        }
+
+        runBlocking {
+            launch {
+                response = client.get("${System.getenv("ETOR_TI_baseurl")}/v1/etor/metadata/" + lookupId) {
+                    authPair.first?.forEach { entry ->
+                        headers.append(entry.key, entry.value)
+                    }
+
+                    headers.append(HttpHeaders.Authorization, "Bearer " + authPair.second!!)
+                }
+                responseBody = response!!.body()
+            }
+        }
+
+        return HttpUtilities.okResponse(request, responseBody)
+    }
+
+    /**
+     * Use the specified report ID to find the lookup ID for the ETOR TI to use to retrieve metadata
+     *
+     * @param id DB Action that we are reviewing
+     * @return the string lookup ID if found, otherwise and empty string
+     */
+    abstract fun getLookupId(id: String): String
```

Comment:
It may be cleaner instead of this abstract method to have `retrieveMetadata` take in a function parameter for this lookup id logic.

Reply:
Also, rename `id` to `reportId` if that's what you are expecting here. Ditto for the other places in this changeset.

Reply:
We've refactored the id parameter name, that's a good catch.  We've observed the established pattern in report stream is to use an abstract function for example `actionIsValid` or `singleDetailedHistory` in SubmissionFunction.  Are you sure you want us to stray from the established pattern?